### DISCUSSION TITLE
Clarify nature of TSP protocol

### DISCUSSION THREAD

**Discussion by Oskar-van-Deventer** [trustoverip/trust-spanning-protocol#14] [General]
What is the nature of the TSP protocol?
-Simplex
-Duplex
-Query-response
-Subscribe-notify
-Session based
-Other

What is the payload of the TSP protocol?
-Only protocols for the exchange of VCs
-Only short messages
-Unlimited-length messages
-Unlimited-size files
-Encoded audio and video
-Any IETF-specified protocol
-Other

What are the performance parameters of the TSP protocol?
-Delay
-Throughput
-Other
-None

It may be good to get and document a common understanding on this, as it is unclear from [section 8 of the Architecture spec](https://github.com/trustoverip/TechArch/blob/main/spec.md#8-the-toip-trust-spanning-protocol).

Oskar

**Comment by sawhitmire** [comment]
The payload and metadata are what we need to define in the TSP. The nature of the connection (simplex, duplex, etc) is the domain of the applications that are communicating. Even the delay and throughput are the domain of applications that sit on top of the TSP. The "right" definition of the TSP is the minimum set of information we need to be able to verify the message and the sender (probably through a separate process).

**Reply by Oskar-van-Deventer** [reply]
So you are saying that, if an application requires a TCP-like connection, then the application should run TCP-over-TSP? And that TSP would be a no-guarantees simplex protocol, like IP?

**Comment by sawhitmire** [comment]
I think I would phrase it as TSP over TCP since TSP would be using TCP to handle routing and connections. There are, or may be, concepts analogous to simplex and duplex and the like at the TSP layer, but I’m not entirely sure yet. Scott WhitmireOn Feb 8, 2023, at 12:18 PM, Oskar van Deventer ***@***.***> wrote:﻿
So you are saying that, if an application requires a TCP-like connection, then the application should run TCP-over-TSP? And that TSP would be a no-guarantees simplex protocol, like IP?

**Comment by talltree** [comment]
@Oskar-van-Deventer As I am finally catching up with discussions I have not been able to reach before, I just wanted to say I think these are good questions that we need to answer in the TSP spec in order to make it clear what is and is not the responsibility of the TSP. So I'm tagging this thread for the #30 Spec Checklist.
